web/src/user_group_edit.js

```diff
@@ -301,49 +364,11 @@ export function handle_member_edit_event(group_id, user_ids) {
     // or remove the group-row on the left panel accordingly.
     const tab_key = get_active_data().$tabs.first().attr("data-tab-key");
     if (tab_key === "your-groups" && user_ids.includes(people.my_current_user_id())) {
-        if (user_groups.is_user_in_group(group_id, people.my_current_user_id())) {
-            // We add the group row to list if the current user
-            // is added to it. The whole list is redrawed to
-            // maintain the sorted order of groups.
-            redraw_user_group_list();
-        } else if (!settings_data.can_join_user_group(group_id)) {
-            // We remove the group row immediately only if the
-            // user cannot join the group again themselves.
-            const group_row = row_for_group_id(group_id);
-            if (group_row.length) {
-                group_row.remove();
-                update_empty_left_panel_message();
-            }
-        }
+        update_your_groups_list(group_id);
     }
 
-    // update display of check-mark.
-    if (is_group_already_present(group)) {
-        const $row = row_for_group_id(group_id);
-
-        const item = group;
-        item.is_member = user_groups.is_user_in_group(group_id, people.my_current_user_id());
-        item.can_join = settings_data.can_join_user_group(item.id);
-        item.can_leave = settings_data.can_leave_user_group(item.id);
-        item.is_direct_member = user_groups.is_direct_member_of(
-            people.my_current_user_id(),
-            item.id,
-        );
-        const associated_subgroups = user_groups.get_associated_subgroups(
-            item,
-            people.my_current_user_id(),
-        );
-        item.associated_subgroup_names =
-            user_groups.group_list_to_comma_seperated_name(associated_subgroups);
-        const html = render_browse_user_groups_list_item(item);
-        const $new_row = $(html);
-
-        // TODO: Remove this if/when we just handle "active" when rendering templates.
-        if ($row.hasClass("active")) {
-            $new_row.addClass("active");
-        }
-
-        $row.replaceWith($new_row);
+    if (user_ids.includes(people.my_current_user_id())) {
```

Comment:
This `if` condition is an addition in this PR, since I'm unable to spot a scenario where we need to update the checkmark with this condition as false.